python/huggingfaceserver/huggingfaceserver/__main__.py

```diff
@@ -69,42 +81,69 @@ def list_of_strings(arg):
     "--return_token_type_ids", action="store_true", help="Return token type ids"
 )
 
-try:
-    from vllm.engine.arg_utils import AsyncEngineArgs
-
-    parser = AsyncEngineArgs.add_cli_args(parser)
-    _vllm = True
-except ImportError:
-    _vllm = False
+parser = maybe_add_vllm_cli_parser(parser)
 
 args, _ = parser.parse_known_args()
 
 if __name__ == "__main__":
     engine_args = None
-    if _vllm and not args.disable_vllm:
+    if args.model_dir:
+        model_id_or_path = Path(Storage.download(args.model_dir))
+    else:
+        model_id_or_path = args.model_id
+
+    if model_id_or_path is None:
+        raise ValueError("You must provide a model_id or model_dir")
+
+    if (
+        vllm_available
+        and not args.disable_vllm
+        and infer_vllm_supported_from_model_architecture(model_id_or_path) is not None
+    ):
+        from .vllm.vllm_model import VLLMModel
+
         args.model = args.model_dir or args.model_id
-        args.revision = args.model_revision
```

Comment:
this is needed for vllm, as vllm uses [revision](https://github.com/vllm-project/vllm/blob/main/vllm/engine/arg_utils.py#L110-L130) as model_revision. Unless we follow the same args here. I am fine on updating it.